src/lib.rs

```diff
@@ -0,0 +1,1210 @@
+            self.truncate(new_len);
+            return
+        }
+        if value && self.len > 0 {
+            // If we're resizing with trues, we need to extend the last block with true bits. We
+            // can rely on mask_last_block to trim any unwanted bits we add in this process.
+            let b = block_offset::<T>(self.len);
+            let v = self.vec[b];
+            self.vec[b] = v | (T::max_value() << (self.len % bits_per_block::<T>()));
+        }
+        let nb = blocks_required::<T>(new_len);
+        self.vec.resize(nb, if value { T::max_value() } else { T::zero() });
+        self.len = new_len;
+        self.mask_last_block();
+    }
+
+    /// Appends all elements in a slice to the Vob.
+    ///
+    /// Iterates over the slice `other` and appends elements in order.
+    ///
+    /// Note that this function is same as extend except that it is specialized to work with slices
+    /// instead. If and when Rust gets specialization this function will likely be deprecated (but
+    /// still available).
+    ///
+    /// # Examples
+    /// ```
+    /// #[macro_use] extern crate vob;
+    /// fn main() {
+    ///     let mut v = vob![true];
+    ///     v.extend_from_slice(&vec![false, true]);
+    ///     assert_eq!(v, vob![true, false, true]);
+    /// }
+    pub fn extend_from_slice(&mut self, other: &[bool]) {
+        for &b in other.iter() {
+            self.push(b);
+        }
+    }
+
+    /// Clears the Vob, removing all values.
+    ///
+    /// Note that this method has no effect on the allocated capacity of the Vob.
+    pub fn clear(&mut self) {
+        self.len = 0;
+        self.vec.clear();
+    }
+
+    /// Sets all bits in the Vob to `value`. Notice that this does not change the number of bits
+    /// stored in the Vob.
+    ///
+    /// # Examples
+    /// ```
+    /// #[macro_use] extern crate vob;
+    /// fn main() {
+    ///     let mut v = vob![true, false, true];
+    ///     v.set_all(false);
+    ///     assert_eq!(v, vob![false, false, false]);
+    /// }
+    /// ```
+    pub fn set_all(&mut self, value: bool) {
+        for b in self.vec.iter_mut() {
+            *b = if value { T::max_value() } else { T::zero() };
+        }
+        self.mask_last_block();
+    }
+
+    /// Negates all bits in the Vob.
+    ///
+    /// # Examples
+    /// ```
+    /// #[macro_use] extern crate vob;
+    /// fn main() {
+    ///     let mut v = vob![true, false];
+    ///     v.negate();
+    ///     assert_eq!(v, vob![false, true]);
+    /// }
+    /// ```
+    pub fn negate(&mut self) {
+        for b in self.vec.iter_mut() {
+            *b = !*b;
+        }
+        self.mask_last_block();
+    }
+
+    /// For each bit in this Vob, `and` it with the corresponding bit in `other`, returning `true`
+    /// if this led to any changes or `false` otherwise. The two Vobs must have the same number of
+    /// bits.
+    ///
+    /// # Panics
+    ///
+    /// If the two Vobs are of different length.
```

Comment:
Is that what we want?

Isn't it common to e.g. `or` a 64-bit value, by say `0x2` (which can be expressed in 1-byte).

Reply:
This is what `BitVec` does and I think it's the right thing. It really doesn't make sense to `or` things of unequal length (what happens with different storage sizes?!).

Reply:
ok.